### PR TITLE
Hint with target ActiveQuery class

### PR DESCRIPTION
**Pull request by bscheshirwork**
```
    /**
     * @return AnotherModelQuery
     */
    public function getAnotherModels()
    {
        return $this->hasMany(AnotherModel::className(), ['anotherId' => 'id'])->inverseOf('firstModel');
    }
```

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | yes/no
| Tests pass?   | yes/no
| Fixed issues  | #287

### CODE REVIEW

**Comment by bscheshirwork** [comment]
up #287

File: generators/model/default/model.php [hunk 1]
```diff
@@ -76,7 +76,25 @@ public function attributeLabels()
 <?php foreach ($relations as $name => $relation): ?>
 
     /**
-     * @return \yii\db\ActiveQuery
+     * @return <?php
+        if ($queryClassName) {
```
Comment (samdark) [inline generators/model/default/model.php]: How about extracting it into model method?

File: generators/model/Generator.php [hunk 2]
```diff
@@ -296,6 +299,37 @@ public function generateLabels($table)
...
+            if ($generateQuery) {
+                $queryClassRealName = '\\' . $this->queryNs . '\\' . $relation[1];
+                if (class_exists($queryClassRealName, true) && is_subclass_of($queryClassRealName, '\yii\db\BaseActiveRecord')) {
+                    /** @var \yii\db\ActiveQuery $activeQuery */
+                    $activeQuery = $queryClassRealName::find();
+                    $activeQueryClass = $activeQuery::className();
+                    if(strpos($activeQueryClass, $this->ns) === 0){
+                        $activeQueryClass = StringHelper::basename($activeQueryClass);
+                    }
+                    $result[$name] = '\yii\db\ActiveQuery|' . $activeQueryClass;
```
Comment (samdark) [inline generators/model/Generator.php]: Why do we need `\yii\db\ActiveQuery|` part here? Isn't `$activeQueryClass` inherited from it already?

File: generators/model/Generator.php [hunk 3]
```diff
@@ -296,6 +299,37 @@ public function generateLabels($table)
...
+            if ($generateQuery) {
+                $queryClassRealName = '\\' . $this->queryNs . '\\' . $relation[1];
+                if (class_exists($queryClassRealName, true) && is_subclass_of($queryClassRealName, '\yii\db\BaseActiveRecord')) {
+                    /** @var \yii\db\ActiveQuery $activeQuery */
+                    $activeQuery = $queryClassRealName::find();
+                    $activeQueryClass = $activeQuery::className();
+                    if (strpos($activeQueryClass, $this->ns) === 0){
+                        $activeQueryClass = StringHelper::basename($activeQueryClass);
+                    }
+                    $result[$name] = '\yii\db\ActiveQuery|' . $activeQueryClass;
```
Comment (samdark) [inline generators/model/Generator.php]: Why do we need `\yii\db\ActiveQuery|` part here? Isn't `$activeQueryClass` inherited from it already?

Comment (bscheshirwork) [reply generators/model/Generator.php]: Mismatch of return type:

`hasMany` returns `\yii\db\ActiveQuery`, not target $activeQueryClass. And another build-in functional don't know about children classes.
Redefine all build-in function each time is not a way.

Comment (samdark) [reply generators/model/Generator.php]: Huh? Isn't inheritance properly handled by IDEs even if we don't override all methods of the class?

Comment (bscheshirwork) [reply generators/model/Generator.php]: I think we don't have a tool for a define child class in PHPDoc of parent class or as return type of parent class 
https://github.com/yiisoft/yii2/blob/9b01ca275f891c245c2e840ed93c87f715c04e70/framework/db/ActiveRecord.php#L72-L73
```php
 * @method ActiveQuery hasMany($class, array $link) see [[BaseActiveRecord::hasMany()]] for more info
 * @method ActiveQuery hasOne($class, array $link) see [[BaseActiveRecord::hasOne()]] for more info
```
not accept `static` or some else like `$this`. I right?.. NO!

Hmmm... I find a solution! :smile_cat:

```php
 * @method $this hasMany($class, array $link) see [[BaseActiveRecord::hasMany()]] for more info
 * @method $this hasOne($class, array $link) see [[BaseActiveRecord::hasOne()]] for more info
```

What about it?

Comment (samdark) [reply generators/model/Generator.php]: Looks good.

Comment (bscheshirwork) [reply generators/model/Generator.php]: So... This is some harder.

I can propose new interface `RelatedActveQueryInterface` and set it in implementation in ActiveRecord

or leave template as is

**Comment by bscheshirwork** [comment]
https://github.com/yiisoft/yii2/issues/14760

File: generators/model/Generator.php (hunk 3 above)
Comment (bscheshirwork) [reply generators/model/Generator.php]: Nope not yet...

This is solution for same class, not another

**Comment by bscheshirwork** [comment]
@samdark no way to avoid "|" :crying_cat_face:

> there is no way to solve this in Yii itself.

**Comment by samdark** [comment]
@bscheshirwork do you plan working more on this one?

**Comment by bscheshirwork** [comment]
"working more"
Anything else needed?
p.s.:merge actual master

File: src/generators/model/default/model.php [hunk 4]
```diff
@@ -76,7 +76,7 @@ public function attributeLabels()
 <?php foreach ($relations as $name => $relation): ?>
 
     /**
-     * @return \yii\db\ActiveQuery
+     * @return <?= $relationsClassHints[$name] . "\n" ?>
```
Comment (rustamwin) [inline src/generators/model/default/model.php]: new line needed?

Comment (samdark) [reply src/generators/model/default/model.php]: Nope.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**